files/acl.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Automattic\VIP\Files\Acl;
+
+const FILE_IS_PUBLIC = 'FILE_IS_PUBLIC';
+const FILE_IS_PRIVATE_AND_ALLOWED = 'FILE_IS_PRIVATE_AND_ALLOWED';
+const FILE_IS_PRIVATE_AND_DENIED = 'FILE_IS_PRIVATE_AND_DENIED';
+const FILE_NOT_FOUND = 'FILE_NOT_FOUND';
+
+function send_visibility_headers( $file_visibility, $file_path ) {
+	// Default to throwing an error so we can catch unexpected problems more easily.
+	$status_code = 500;
+	$header = false;
+
+	switch ( $file_visibility ) {
+		case FILE_IS_PUBLIC:
+			$status_code = 202;
+			break;
+
+		case FILE_IS_PRIVATE_AND_ALLOWED:
+			$status_code = 202;
+			$header = 'X-Private: true';
+			break;
+
+		case FILE_IS_PRIVATE_AND_DENIED:
+			$status_code = 403;
+			$header = 'X-Private: true';
+			break;
+
+		case FILE_NOT_FOUND:
+			// This is a signal for the server to pass the request through to the Files Service.
+			// Even though the attachment was not found within WordPress, it may still be a physical file in the Files Service.
+			$status_code = 202;
```

Comment:
This behavior seems counterintuitive and may turn into a "gotcha" in the future. If we're giving control to the application to say a file is not found...then shouldn't we honor that and not attempt to serve anything? Otherwise, we do we let the application set this code?

My worry is that folks will try to "hide" the existence of certain files by assuming they can return the FILE_NOT_FOUND code...but that will actually leave the file open to being served by the file service, which will not be expected behavior.

Reply:
That's a good point.

The particular thinking here was for unpublished media (https://gist.github.com/mjangda/8657cfde2b2a4cafd95f509dd1d3dde9#file-vip-private-files-unpublished-media-php). Specifically, what to do when the file doesn't exist as an attachment in WP. But I think it probably makes sense just to drop this and return this status and return as `FILE_IS_PUBLIC`. We can add the logic later if needed.

Reply:
Removed in 5f2972db4